src/app/shared/autocomplete/autocomplete.component.ts

```diff
@@ -0,0 +1,54 @@
+import {Component, OnDestroy, OnInit, Output, EventEmitter, Input} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {Subscription} from 'rxjs';
+
+@Component({
+  selector: 'app-autocomplete',
+  templateUrl: './autocomplete.component.html',
+  styleUrls: ['./autocomplete.component.less']
+})
+export class AutocompleteComponent implements OnInit, OnDestroy {
+  @Input() placeholder: string;
+  @Input() sourceArray: any[];
```

Comment:
I believe we should use a strict interface for source array.
For example:
[
{
 label: string;
 value: string;
}
]